rtCamp/web-stories-wp
# Feature/deprecate embed block

## Context

With the addition of the new Web Stories block, the existing Stories Embed block needs to be deprecated and handle the block transformation.

## Summary

This PR removes the Stories Embed block and adds appropriate transform and deprecation definition for the new block to handle older content.

- Technically this PR updates the Stories Embed block with new block types, latest and selected stories.
- These newly added block types enables users to embed multiple stories on a page.

## Relevant Technical Choices

- Removes existing files relating to the Stories Embed block.
- Renames Web Stories List to Stories Embed block.

## To-do

- Update any relevant documentation for the Stories block.
- Expand test coverage for the new block.

## User-facing changes

- Users may observe changes in older embed block which now will have transformed into newer block.

## Testing Instructions

- Test new web stories block like any other block, add it to any post and try different settings.

Test with older content.
- Create a new post with older block content, either using the classic editor's 'text' view or block editors 'Code editor' and save the post. Following is the snippet of older block.

```

<div class="wp-block-web-stories-embed alignnone"><a href="https://preview.amp.dev/documentation/examples/introduction/stories_in_amp"><img src="https://amp.dev/static/samples/img/story_dog2_portrait.jpg" alt="Stories in AMP - Hello World" width="360" height="600" /></a></div>

```

- Now, switch to the block editor and refresh once.
- Check that the block toolbar now has a transform button to switch between blocky types.

Test 'Convert to blocks' with classic block and shortcode.
- Add following shortcode to a classic block.
```
[web_stories_embed align="none" height="457" width="285" poster="https://amp.dev/static/samples/img/story_dog2_portrait.jpg" title="Test Story" url="https://preview.amp.dev/documentation/examples/introduction/stories_in_amp" /]
```
- Click the button which says 'Convert to blocks' on the classic block toolbar.
- This should transform the shortcode to web stories block without any errors.

Test content added in classic editor.
- Use [Classic Editor](https://wordpress.org/plugins/classic-editor/) plugin for this test.
- Create new post with older content, use above provided content.
- Now open the same post with block editor and use the 'Convert to blocks' on the classic content.
- This should transform the content to new block, without any errors.


### UAT
